### PR TITLE
Always infer dependencies in build-script

### PR DESCRIPTION
**Pull request by drexin**
As preparation to move products from build-script-impl to build-script, we need to always enable dependency inference.

### CODE REVIEW

**Comment by drexin** [comment]
@swift-ci smoke test

**Comment by drexin** [comment]
@swift-ci smoke test

**Comment by drexin** [comment]
@swift-ci smoke test

File: utils/swift_build_support/tests/products/test_cmark.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
+# tests/products/test_ninja.py ----------------------------------*- python -*-
```
Comment (drexin) [inline utils/swift_build_support/tests/products/test_cmark.py:1]: Oops, this shouldn't have been committed.

File: utils/build-script [hunk 2]
```diff
@@ -978,7 +968,7 @@ class BuildScriptInvocation(object):
                     product.test(host_target)
                     print("--- Finished tests for %s ---" % product_name)
                 if product.should_install(host_target) or \
-                   (self.install_all and product.should_build(host_target)):
+                   product.should_build(host_target):
```
Comment (gottesmm) [inline utils/build-script:971]: The whitespace here is weird.

**Review by gottesmm — APPROVED** [review]
LGTM. My only concern is do we do this even with presets? Or do we treat presets as specifying things specifically?

I am fine with doing this for presets as well, I just think that we need to be thorough around the testing and make sure we aren't breaking any bots by mistake.

**Comment by drexin** [comment]
@gottesmm Yeah, I'm still trying to find the best way to do it. Maybe enabling it by default and disabling it via flag is the better option for now? I.e. turn `--infer` (toggle_true) into `--no-infer` (toggle_false)?

**Pull request closed without merging**